### ISSUE TITLE
TypeError: argument of type 'NoneType' is not iterable

### ISSUE THREAD

**Issue by mysystem32** [jasonacox/tinytuya#68]
Firstly, thanks for tinytuya - really useful.

Getting an stack trace when reading status from a DDS238-2 wifi smart energy meter.
```
Traceback (most recent call last):
  File "/home/plug/plug.py", line 401, in <module>
    data = d.status()
  File "/usr/lib/python3.9/site-packages/tinytuya/__init__.py", line 904, in status
    if 'Err' in data:
TypeError: argument of type 'NoneType' is not iterable

# Offending code:
 892 class Device(XenonDevice):
 893     def __init__(self, dev_id, address, local_key="", dev_type="default"):
 894         super(Device, self).__init__(dev_id, address, local_key, dev_type)
 895
 896     def status(self):
 897         """Return device status."""
 898         log.debug('status() entry (dev_type is %s)', self.dev_type)
 899         payload = self.generate_payload(DP_QUERY)
 900
 901         data = self._send_receive(payload)
 902         log.debug('status() received data=%r', data)
 903         # Error handling
 904         if 'Err' in data:
 905             if data['Err'] == str(ERR_DEVTYPE):
 906                 # Device22 detected and change - resend with new payload
 907                 log.debug('status() rebuilding payload for device22')
 908                 payload = self.generate_payload(DP_QUERY)
 909                 data = self._send_receive(payload)
 910
 911         return data
The received data is None
Change 904 to
if dps and 'Err' in data:
```
This fixed the problem for me.
I do check the result in my app for None and retry.

**Comment by jasonacox** [comment]
Hi @mysystem32  - thank you!  Just to confirm, did you set line 904 to:

```python
if data and 'Err' in data:
```

The reason I ask, 'dps' is not a variable in scope but I think you mean 'data'?

**Comment by mysystem32** [comment]
Apologies,

yes data

--
***@***.***
+27-83-391-8986

**Comment by mysystem32** [comment]
The reason I get None is that the smart plug is in a metal box that limits
WiFi reception.

--
***@***.***
+27-83-391-8986

**Comment by jasonacox** [comment]
Hi @mysystem32 - I have released v1.2.9 - https://pypi.org/project/tinytuya/ - that now contains the logic to handle this condition.  It should be available via `pip install --upgrade tinytuya` soon.  Thanks for all your help with this.  I'll close this issue but we can re-open if you have other issues.  Thanks!!

**Comment by mysystem32** [comment]
Thanks!!!

--
***@***.***
+27-83-391-8986
